colearn_grpc/utils.py

```diff
@@ -0,0 +1,80 @@
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2021 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+import pickle
+from math import ceil
+from typing import Iterator
+
+from colearn.ml_interface import Weights
+from colearn_grpc.proto.generated.interface_pb2 import WeightsPart
+
+WEIGHTS_PART_SIZE_BYTES = 4 * 10 ** 6
```

Comment:
where this number is coming from?

Reply:
The default limit for gRPC messages is 4MB, so this number is a bit less. It's fairly arbitrary!